src/i_sdlsound.c

```diff
@@ -412,26 +414,38 @@ static boolean ExpandSoundData_SRC(sfxinfo_t *sfxinfo,
     int16_t *expanded;
     allocated_sound_t *snd;
     Mix_Chunk *chunk;
+    uint32_t samplecount = length / (bits / 8);
 
-    src_data.input_frames = length;
-    data_in = malloc(length * sizeof(float));
+    src_data.input_frames = samplecount;
+    data_in = malloc(samplecount * sizeof(float));
     src_data.data_in = data_in;
     src_data.src_ratio = (double)mixer_freq / samplerate;
 
     // We include some extra space here in case of rounding-up.
-    src_data.output_frames = src_data.src_ratio * length + (mixer_freq / 4);
+    src_data.output_frames = src_data.src_ratio * samplecount + (mixer_freq / 4);
     src_data.data_out = malloc(src_data.output_frames * sizeof(float));
 
     assert(src_data.data_in != NULL && src_data.data_out != NULL);
 
     // Convert input data to floats
 
-    for (i=0; i<length; ++i)
+    if (bits == 16)
```

Comment:
This block, for example, should get a comment.

Reply:
https://github.com/fabiangreffrath/crispy-doom/pull/511/commits/69004443395e6d745cee282866fd48e0bd2bf11e